sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/TracerSdkProvider.java

```diff
@@ -36,7 +36,7 @@
  * io.opentelemetry.OpenTelemetry}. However, if you need a custom implementation of the factory, you
  * can create one as needed.
  */
-public class TracerSdkProvider implements TracerProvider {
+public class TracerSdkProvider implements TracerProvider, TracerSdkManagement {
```

Comment:
Does the new interface allow this to be private? And then maybe we don't need to change the return types

Reply:
what do you want to be private?

Reply:
If you want to make TracerSdkProvider package-private, we would need to move the SPI implementation into this package: `TracerProviderFactorySdk`. That seems possible, if we're ok with moving it.

Reply:
although it's used in a lot of tests as well right now, so we'd need to do quite a bit of test refactoring.

Reply:
I feel like moving the SPI implementation into this package is a win, less API is nice - I would say SPI package usually defines an SPI interface, but doesn't have to or even should define an SPI implementation. Doesn't have to be in this PR though.